src/state/error.tsx

```diff
@@ -0,0 +1,23 @@
+import { createContext, PropsWithChildren, useState } from 'react'
+
+export const ErrorContext = createContext<{
```

Comment:
Instead of exporting the `ErrorContext`, export a getter/setter:

    export function useError() {
      return useContext(ErrorContext).error
    }
    
    export function useSetError() {
      return useContext(ErrorContext).setError
    }